django/db/models/base.py

```diff
@@ -569,6 +570,15 @@ def _get_pk_val(self, meta=None):
         return getattr(self, meta.pk.attname)
 
     def _set_pk_val(self, value):
+        for parent_link, (parent, field) in zip(
+                self._meta.parents.values(),
+                self._meta.parents.items()
+                ):
+            if parent_link and field:
```

Comment:
IMO this check is unnecessary.

Reply:
So for instance, if `self._meta.parents` contains a non-null key and null (None) value pair, `parent_link` will expand as `None` and `parent_link.target_field` will throw an error. This happened for me in "test_proxy_inheritance_supported" under "bulk_create.tests.BulkCreateTests"  during local testing. Hence I put that check here.

Reply:
Agreed, thanks.